### ISSUE TITLE
Add snapshot / checkpoint ACs for the referral program

### ISSUE THREAD

**Issue by gordsport** [vegaprotocol/specs#1980]
Having looked at the ACs for the referral specs:

- [RFPR](https://github.com/vegaprotocol/specs/blob/cosmicelevator/protocol/0083-RFPR-on_chain_referral_program.md)
- [VDPR](https://github.com/vegaprotocol/specs/blob/cosmicelevator/protocol/0084-VDPR-volume_discount_program.md)
- [RVST](https://github.com/vegaprotocol/specs/blob/cosmicelevator/protocol/0085-RVST-rewards_vesting.md)
- [ASPR](https://github.com/vegaprotocol/specs/blob/cosmicelevator/protocol/0086-ASPR-activity_streak_program.md)

There is nothing to cover what should / should not be captured in snapshots or checkpoints. 

These ACs should be added and covered in system tests

**Comment by wwestgarth** [comment]
I don't think we need an explicit AC for snapshots. What I think is missing is some nice "happy-path" ACs that can be implemented in the system test.

For context: I recently fixed a snapshot bug on fairground related to referrals, and wondered how it wasn't caught earlier in our automatic-snapshot-testing on the sytem test pipelines. The reason looks to be because we've shifted a lot of the functional testing around referrals/fees/discounts to the feature test. The ones pegged as system-tests are more basic proposal validation/API things, and less functional.

So what I think is missing is a couple of ACs that can be implemented in the system tests that flex the most _basic_ happy path. Something like "a party aggressively trades this volume, and their fees will be less because of the volume discount", or "they are part of this referral set, and do this, so they have X in their vesting rewards". Just so we have at the system test level at least one test of the most basic end-to-end scenario for each feature.
